openmetadata-service/src/main/java/org/openmetadata/service/util/S3Util.java

```diff
@@ -0,0 +1,109 @@
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.openmetadata.service.util;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import org.apache.logging.log4j.util.Strings;
+import org.openmetadata.schema.entity.data.Table;
+import org.openmetadata.schema.services.connections.database.SampleDataS3Config;
+import org.openmetadata.schema.type.TableData;
+import org.openmetadata.service.secrets.SecretsManagerFactory;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.ResponseInputStream;
+import software.amazon.awssdk.core.sync.RequestBody;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.GetObjectRequest;
+import software.amazon.awssdk.services.s3.model.PutObjectRequest;
+import software.amazon.awssdk.services.s3.model.S3Exception;
+
+public class S3Util {
+
+  S3Client client;
+
+  private String getOrDefault(String original, String defaultStr) {
+    return Strings.isNotEmpty(original) ? original : defaultStr;
+  }
+
+  private void initClient(SampleDataS3Config s3Config) {
+    if (s3Config != null
+        && s3Config.getAwsConfig() != null
+        && Strings.isNotEmpty(s3Config.getAwsConfig().getAwsRegion())) {
+      SecretsManagerFactory.getSecretsManager().decryptSampleDataS3Config(s3Config);
+      String region = s3Config.getAwsConfig().getAwsRegion();
+      String accessKeyId = this.getOrDefault(s3Config.getAwsConfig().getAwsAccessKeyId(), "");
+      String secretAccessKey = this.getOrDefault(s3Config.getAwsConfig().getAwsSecretAccessKey(), "");
+      AwsCredentialsProvider credentialsProvider;
+      if (Strings.isBlank(accessKeyId) && Strings.isBlank(secretAccessKey)) {
+        credentialsProvider = DefaultCredentialsProvider.create();
+      } else {
+        credentialsProvider =
+            StaticCredentialsProvider.create(AwsBasicCredentials.create(accessKeyId, secretAccessKey));
+      }
+      this.client = S3Client.builder().region(Region.of(region)).credentialsProvider(credentialsProvider).build();
+    } else {
+      this.client = S3Client.create();
+    }
+  }
+
+  private String getSampleDataObjectKey(SampleDataS3Config s3Config, Table table) {
+    return getOrDefault(s3Config.getPrefix(), "")
+        + "/"
+        + table.getService().getName()
+        + "/"
+        + table.getDatabase().getName()
+        + "/"
+        + table.getDatabaseSchema().getName()
+        + "/"
+        + table.getName()
+        + "/sample_data.json";
```

Comment:
this should be a parquet

Reply:
and maybe we can `"/".join(...)` with the Java equivalent. Also, not sure if we have the `/` defined as a separator anywhere, otherwise we could

Reply:
Sure, can do that.